src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 1 : 0)
```

Comment:
This code seems kind of convoluted, but more importantly, would not the double negative here cause the set of realistic values to be [0, 1] ?  

Suggest:
```javascript
const shift =  takerGetsCurrency === 'XRP' ? 6 : 
               takerPaysCurrency === 'XRP' ? -6 :  0;

return shift ? (new BigNumber(quality)).shift(shift).toString() : 
               quality;
```

Maybe a test?